packages/core/lib/get-permissions.ts

```diff
@@ -1,28 +1,73 @@
 import {
+  AppState,
   ComponentData,
   Config,
   DefaultComponentProps,
   Permissions,
 } from "../types/Config";
+import { getChanged } from "./get-changed";
+
+import { resolvePermissions } from "./resolve-permissions";
+
+const cache: {
+  lastPermissions: Record<string, Partial<Permissions>>;
+  lastSelected: ComponentData | {} | undefined;
+} = { lastPermissions: {}, lastSelected: {} };
 
 export const getPermissions = ({
   selectedItem,
   type,
   globalPermissions,
   config,
+  appState,
 }: {
   selectedItem?: ComponentData | undefined;
   type?: keyof DefaultComponentProps;
   globalPermissions: Partial<Permissions>;
   config: Config;
+  appState: AppState;
 }) => {
   const componentType = type || (selectedItem && selectedItem.type) || "";
+  const componentId = (selectedItem && selectedItem.props.id) || "";
+
   let componentPermissions = getInitialPermissions({
     componentType,
     config,
     globalPermissions,
   });
 
+  const changed = getChanged(selectedItem, cache.lastSelected);
+
+  if (Object.values(changed).some((el) => el === true)) {
+    const resolvedPermissions = resolvePermissions({
```

Comment:
I think this implementation works for now as it's synchronous

I know we talked a lot about synchronous vs asynchronous here and landed in synchronous to keep it more simple, but when implementing the async resolver, we will likely need to consider when this is called to avoid triggering `resolvePermissions` multiple times in parallel.

I'll open a separate ticket for async resolvers, and will aim to look at it later this week.

Reply:
Tracking: #582